src/templateNoDistractingElementsRule.ts

```diff
@@ -0,0 +1,51 @@
+import { ElementAst } from '@angular/compiler';
+import { sprintf } from 'sprintf-js';
+import { IRuleMetadata, RuleFailure, Rules } from 'tslint/lib';
+import { SourceFile } from 'typescript/lib/typescript';
+import { NgWalker } from './angular/ngWalker';
+import { BasicTemplateAstVisitor } from './angular/templates/basicTemplateAstVisitor';
+
+export class Rule extends Rules.AbstractRule {
+  static readonly metadata: IRuleMetadata = {
+    description: 'Enforces that no distracting elements are used',
+    options: null,
+    optionsDescription: 'Not configurable.',
+    rationale: 'Elements that can be visually distracting can cause accessibility issues with visually impaired users.',
+    ruleName: 'template-no-distracting-elements',
+    type: 'functionality',
+    typescriptOnly: true
+  };
+
+  static readonly FAILURE_STRING = 'Avoid using <%s/> elements as they create visual accessibility issues.';
+
+  apply(sourceFile: SourceFile): RuleFailure[] {
+    return this.applyWithWalker(
+      new NgWalker(sourceFile, this.getOptions(), {
+        templateVisitorCtrl: TemplateNoDistractingElementsVisitor
+      })
+    );
+  }
+}
+
+export function getFailureMessage(element: string) {
+  return sprintf(Rule.FAILURE_STRING, element);
+}
+
+class TemplateNoDistractingElementsVisitor extends BasicTemplateAstVisitor {
+  visitElement(prop: ElementAst, context: any): any {
+    this.validateElement(prop);
+    super.visitElement(prop, context);
+  }
+
+  private validateElement(el: ElementAst): void {
+    if (el.name === 'marquee' || el.name === 'blink') {
```

Comment:
When I see [this link](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/blink#Browser_compatibility), I'm not sure that we have to test 'blink'.

Reply:
I think its fine, since all the accessibility checkers and linters check for it. Sometimes developers use code generated or from old references.